crates/sequencer/src/state.rs

```diff
@@ -16,9 +22,235 @@ use starknet_api::{
 /// See [rustc-hash](https://crates.io/crates/rustc-hash) for more information.
 #[derive(Default)]
 pub struct State {
-    pub classes: FxHashMap<ClassHash, ContractClass>,
-    pub compiled_classes: FxHashMap<ClassHash, CompiledClassHash>,
-    pub contracts: FxHashMap<ContractAddress, ClassHash>,
-    pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
-    pub nonces: FxHashMap<ContractAddress, Nonce>,
+    classes: FxHashMap<ClassHash, ContractClass>,
+    compiled_class_hashes: FxHashMap<ClassHash, CompiledClassHash>,
+    contracts: FxHashMap<ContractAddress, ClassHash>,
+    storage: FxHashMap<ContractStorageKey, StarkFelt>,
+    nonces: FxHashMap<ContractAddress, Nonce>,
+}
+
+/// State implementation for the sequencer. We use a mutable reference to the state
+/// because this is what will be available during the implementation of the execution.
+impl<'a> BlockifierState for &'a mut State {
```

Comment:
why do we need lifetimes in our context?

Reply:
we provide lifetime in the impl definition, but never inside the functions below

Reply:
I guess you are right, I don't need the lifetime here, my bad